lib/valueflow.cpp

```diff
@@ -1658,8 +1658,79 @@ static bool evalAssignment(ValueFlow::Value &lhsValue, const std::string &assign
     return true;
 }
 
+template<class T>
+struct SingleRange {
+    T* x;
+    T* begin() const {
+        return x;
+    }
+    T* end() const {
+        return x+1;
+    }
+};
+
+template<class T>
+SingleRange<T> MakeSingleRange(T& x)
+{
+    return {&x};
+}
+
+class MultiRange {
+    using M = std::unordered_map<nonneg int, ValueFlow::Value>;
+
+    struct Iterator {
+        using value_type = const ValueFlow::Value;
+        using pointer = value_type *;
+        using reference = value_type &;
+
```

Comment:
This should have `using iterator_category = forward_iterator_tag` typedef.

Reply:
done

Reply:
Would this be something for Cppcheck to warn about?

Reply:
hmm.. maybe I am not strongly against it.